### PR TITLE
feat: add support for api keys

### PR DESCRIPTION
**Pull request by gergof**
Closes: elastic/apm-agent-nodejs#1693

Added support for using api keys to authenticate to the APM server.
This PR needs the following PR to be merged first: https://github.com/elastic/apm-nodejs-http-client/pull/110

### Checklist

- [x] Implement code
- [x] Add tests
- [x] Update TypeScript typings
- [x] Update documentation
- [x] Commit message follows [commit guidelines](https://github.com/elastic/apm-agent-nodejs/blob/master/CONTRIBUTING.md#commit-message-guidelines)

### CODE REVIEW

**Comment by astorm** [comment]
jenkins run the tests please

**Comment by astorm** [comment]
jenkins run the tests please

File: lib/config.js [hunk 1]
```diff
@@ -81,6 +81,7 @@ var ENV_TABLE = {
   abortedErrorThreshold: 'ELASTIC_APM_ABORTED_ERROR_THRESHOLD',
   active: 'ELASTIC_APM_ACTIVE',
   addPatch: 'ELASTIC_APM_ADD_PATCH',
+  apiKey: 'ELASTIC_APM_API_KEY',
   apiRequestSize: 'ELASTIC_APM_API_REQUEST_SIZE',
```
Comment (astorm) [inline lib/config.js:85]: 👍 I see here that we're adding the ENV variable `ELASTIC_APM_API_KEY`, which matches what the other agent teams have done and is the [agreed upon spec behavior](https://github.com/elastic/apm/issues/183).

File: lib/config.js [hunk 2]
```diff
@@ -254,6 +255,7 @@ class Config {
 
           // HTTP conf
           secretToken: conf.secretToken,
+          apiKey: conf.apiKey,
           userAgent: userAgent,
```
Comment (astorm) [inline lib/config.js:259]: I see here we're setting `apiKey` in the configuration directly from the value set in the agent configuration.  This means we're expecting users to configure the base64 encoded value of the `id:key` pair (i.e. we don't base64 encode it for them) This is [the correct specified behavior](https://github.com/elastic/apm/issues/183).

**Review by astorm — COMMENTED** [review]
@gergof Looks like a solid contribution, thank you!  I made some review comments about expanding the docs slightly, and I'm going to put this out there for wider review.   If you can get those doc changes landed I'll give this the green approval check.

File: docs/custom-stack.asciidoc [hunk 3]
```diff
@@ -48,6 +48,9 @@ var apm = require('elastic-apm-node').start({
   // Use if APM Server requires a token
   secretToken: '',
 
+  // Use if APM Server uses API keys for authentication
+  apiKey: '',
+
```
Comment (trentm) [inline docs/custom-stack.asciidoc:53]: Not required, just throwing this out there. Could combine secretToken and apiKey into something like.

```
  // If the APM Server requires authentication via a token or API key use one of:
  secretToken: '',
  apiKey: '',
```

**Review by trentm — COMMENTED** [review]
Shall we do a 9.4.2 release of the http client so that we can specify that in package.json here?

Then the TODOs will be:

- [ ] update package.json for "^9.4.2"
- [ ] add a note in the `[[release-notes-x.x.x]]` section of CHANGELOG.asciidoc
- [ ] improve the configuration.asciidoc section

**Comment by trentm** [comment]
elastic-apm-http-client@9.4.2 is published to npm now.

**Comment by astorm** [comment]
jenkins run the tests

**Comment by astorm** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by astorm** [comment]
@gergof 👋 howdy -- this PR looks mostly ready to go. There's just a few final i's to dot and T's to cross (@trentm outlined them here: https://github.com/elastic/apm-agent-nodejs/pull/1818#pullrequestreview-537657791). Are you still-interested-in (and have time for) updating the http client's version in `package.json` to "^9.4.2", adding a note to the `CHANGELOG.asciidoc` file and making the edits to `configuration.asciidoc` as requested in the review? 

If not, no worries -- we can tackle that for you.  Just let us know if we should step on your toes or not ;) 

Thanks again for the PR, and best wishes.

**Comment by gergof** [comment]
I'm resolving it today

**Comment by astorm** [comment]
jenkins run the tests

**Comment by astorm** [comment]
jenkins run the tests

**Comment by gergof** [comment]
I'm done with the requested changes and the branch is up-to-date with the master

**Comment by astorm** [comment]
jenkins run the tests

**Comment by trentm** [comment]
jenkins run the tests

**Comment by astorm** [comment]
jenkins run the tests

**Comment by astorm** [comment]
jenkins run the tests

**Comment by astorm** [comment]
jenkins run the tests please

**Comment by astorm** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by astorm** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Comment by astorm** [comment]
jenkins run the tests please

**Comment by astorm** [comment]
@elasticmachine, run elasticsearch-ci/docs

**Review by astorm — APPROVED** [review]
@gergof thank you for both your work and persistence on this one.  I've done a final review/check and was able to successfully authenticate using a base64 encoded id/key pair created via `POST /_security/api_key` [as outlined in the APM Server Docs](https://www.elastic.co/guide/en/apm/server/master/api-key.html#create-api-key)

```
    {
        /* ... */,
        apiKey: Buffer.from(`${id}:${api_key}`).toString('base64')
        /* ... */,        
    }
```    

I was also able to use an already encoded base64 key as returned by APM Server's `apiKey` command (`Credentials` below)

    % ./apm-server apikey create --ingest --agent-config --name 'test key apm server'
    API Key created:

    Name ........... temp key apm server
    Expiration ..... never
    Id ............. S_Zwc3YBS0I5iIMkOZ8H
    API Key ........ H...w (won't be shown again)
    Credentials .... U...w==

Tests all look like they're passing, so we are good to go. Approving/merging.

**Pull request merged**